java/src/main/java/ai/rapids/cudf/CuFileBuffer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.rapids.cudf;
+
+/**
+ * Represents a cuFile buffer.
+ */
+public final class CuFileBuffer extends BaseDeviceMemoryBuffer {
+  private static final int ALIGNMENT = 4096;
+
+  private final CuFileResourceCleaner cleaner;
+
+  /**
+   * Construct a new cuFile buffer.
+   *
+   * @param buffer         The device memory buffer used for the cuFile buffer.
+   * @param registerBuffer If true, register the cuFile buffer.
+   */
+  public CuFileBuffer(DeviceMemoryBuffer buffer, boolean registerBuffer) {
```

Comment:
Who is responsible for closing `buffer`?  The Javadoc isn't clear whether this class is expected to own the buffer (and if so, where is that buffer being freed?) or if the caller is responsible for keeping the buffer valid as long as this instance is alive.  If this class is not expected to own it but the buffer is expected to stay valid for the life of this instance, it might be best to increment the refcount of the buffer in this constructor.  Then whether it owns or not, this class would close `buffer` when it is closed.

Reply:
Ok, changed the `CuFileBuffer` to own the underlying `DeviceMemoryBuffer`. The client code looks cleaner. Please take another look.